src/client/Microsoft.Identity.Client/Credential/CertificateCache.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Credential
+{
+    internal class CertificateCache
+    {
+        private static readonly Lazy<CertificateCache> s_instance = new Lazy<CertificateCache>(() => new CertificateCache());
+        private X509Certificate2 _cachedCertificate;
+        private readonly object _certificateLock = new();
+
+        private CertificateCache()
+        {
+        }
+
+        public static CertificateCache Instance() => s_instance.Value;
```

Comment:
Generally avoid singletons. Might as well be all static. 
If a class needs to be mocked, use interfaces.

Reply:
deleted the cert cache